great_expectations/experimental/datasources/interfaces.py

```diff
@@ -398,7 +414,7 @@ def get_asset(self, asset_name: str) -> _DataAssetT:
                 f"'{asset_name}' not found. Available assets are {list(self.assets.keys())}"
             ) from exc
 
-    def add_asset(self, asset: _DataAssetT) -> _DataAssetT:
+    def _add_asset_and_test_connection(self, asset: _DataAssetT) -> _DataAssetT:
```

Comment:
This old `add_asset` method felt like it should be a private method that users should never need to call.